### PR TITLE
admin manage users

### PR DESCRIPTION
**Pull request by VuongLong**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Admin::UsersController < ApplicationController
+  def index
+    @users = User.paginate page: params[:page]
```
Comment (phamkykhoi) [inline app/controllers/admin/users_controller.rb]: Insert per_page: Settings

File: app/views/admin/users/_user.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<tr>
+  <td><%= link_to user.name, user %></td>
+  <td><%= link_to "delete", admin_user_path(user), method: :delete,
+                                  data: {confirm: t("admin.delete_confirm")} %></td>
+  <td>
+  	<% if user.admin? %>
```
Comment (phamkykhoi) [inline app/views/admin/users/_user.html.erb]: Indent

File: app/views/admin/users/_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  	  <%= form_for [:admin, user] do |f| %>
```
Comment (phamkykhoi) [inline app/views/admin/users/_user.html.erb]: Indent

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<% provide :title, t("admin.all_user") %>
+<h1>All users</h1>
+<table class="table">
+  <tr>
+	  <th><%= t("admin.table.user") %></th>
```
Comment (phamkykhoi) [inline app/views/admin/users/index.html.erb]: Indent

File: app/views/admin/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<% provide :title, t("admin.all_user") %>
+<h1>All users</h1>
```
Comment (phamkykhoi) [inline app/views/admin/users/index.html.erb]: I18n

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  def update
+    @user = User.find params[:id]
+    if @user.update_attributes user_params
+      redirect_to admin_root_url
+    else
+      redirect_to admin_root_url
+    end
+  end
+
+  def destroy
```
Comment (phamkykhoi) [inline app/controllers/admin/users_controller.rb]: Em nên viết một action set_user trong private để dùng chung cho các action edit, destroy, show

File: app/views/admin/users/_user.html.erb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<tr>
+  <td><%= link_to user.name, user %></td>
+  <td><%= link_to "delete", admin_user_path(user), method: :delete,
+                                  data: {confirm: t("admin.delete_confirm")} %></td>
+  <td>
+    <% if user.admin? %>
+      <%= form_for [:admin, user] do |f| %>
+        <%= f.hidden_field :admin, value: false %>
+        <%= f.submit t("admin.set_user_normal"), class: "btn btn-primary" %>
+      <% end %>
```
Comment (phamkykhoi) [inline app/views/admin/users/_user.html.erb]: Indent

File: config/routes.rb [hunk 8]
```diff
@@ -9,4 +9,8 @@
   delete  "logout"    =>   "sessions#destroy"
   resources :users
   resources :relationships, only: [:create, :destroy]
+  namespace :admin do
+  	root "users#index"
```
Comment (phamkykhoi) [inline config/routes.rb]: Indent

File: config/routes.rb [hunk 9]
```diff
@@ -9,4 +9,8 @@
...
+  	resources :users
```
Comment (phamkykhoi) [inline config/routes.rb]: Indent

File: Gemfile [hunk 10]
```diff
@@ -5,6 +5,8 @@ gem "sass-rails", "~> 5.0"
 gem "uglifier", ">= 1.3.0"
 gem "coffee-rails", "~> 4.1.0"
 gem "bootstrap-sass",       "3.2.0.0"
+gem 'will_paginate',           "3.0.7"
+gem 'bootstrap-will_paginate', "0.0.10"
```
Comment (dieunb) [inline Gemfile]: `"` instead of `'`

File: app/controllers/admin/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def edit
+  end
+
+  def update
+    if @user.update_attributes user_params
+      redirect_to admin_root_url
+    else
+      redirect_to admin_root_url
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: `redirect_to admin_root_url` is same above. You should refactor it and add flash message

**Pull request merged**